sphinxgallery/gen_rst.py

```diff
@@ -292,157 +254,175 @@ def scale_image(in_fname, out_fname, max_width, max_height):
 
-def generate_file_rst(fname, target_dir, src_dir, plot_gallery):
+
+def generate_file_rst(fname, target_dir, src_dir):
     """ Generate the rst file for a given example."""
-    base_image_name = os.path.splitext(fname)[0]
-    image_fname = 'sphx_glr_%s_%%03d.png' % base_image_name
 
-    this_template = rst_template
-    short_fname = target_dir.replace(os.path.sep, '_') + '_' + fname
     src_file = os.path.join(src_dir, fname)
     example_file = os.path.join(target_dir, fname)
     shutil.copyfile(src_file, example_file)
 
     image_dir = os.path.join(target_dir, 'images')
-    thumb_dir = os.path.join(image_dir, 'thumb')
-    thumb_file = os.path.join(thumb_dir, 'sphx_glr_%s_thumb.png' % base_image_name)
     if not os.path.exists(image_dir):
         os.makedirs(image_dir)
-    if not os.path.exists(thumb_dir):
-        os.makedirs(thumb_dir)
+
+    base_image_name = os.path.splitext(fname)[0]
+    image_fname = 'sphx_glr_' + base_image_name + '_{0:03}.png'
+    image_path = os.path.join(image_dir, image_fname)
+
+    if _plots_are_current(src_file, image_path):
+        return
 
     time_elapsed = 0
-    if plot_gallery and fname.startswith('plot'):
-        # generate the plot as png image if file name
-        # starts with plot and if it is more recent than an
-        # existing image.
-        image_list, time_elapsed, stdout = execute_script(image_dir,
-                                                          thumb_file,
-                                                          image_fname,
-                                                          base_image_name,
-                                                          src_file, fname)
-        this_template = plot_rst_template
-
-    if not os.path.exists(thumb_file):
-        # create something to replace the thumbnail
-        scale_image(os.path.join(glr_path_static(), 'no_image.png'),
-                    thumb_file, 200, 140)
+    script_blocks = split_code_and_text_blocks(example_file)
+
+    ref_fname = example_file.replace(os.path.sep, '_')
+    example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
+
+    if not fname.startswith('plot'):
+        convert_func = dict(code=codestr2rst, text=text2string)
+        for blabel, bcontent in script_blocks:
+            example_rst += convert_func[blabel](bcontent)+'\n'
+    else:
+        example_globals = {}
+        fig_count = 0
+        for blabel, bcontent in script_blocks:
+            if blabel == 'code':
+                code_output, rtime, fig_count = execute_script(bcontent,
+                                                               example_globals,
+                                                               image_path,
+                                                               fig_count,
+                                                               src_file)
+
+                time_elapsed += rtime
+                # Single example style
+                if len(script_blocks) == 2:
```

Comment:
script_blocks length is not going to change in the `for blabel, bcontent in script_blocks` loop. You should put that in a nicely named boolean variable outside of the loop, for example:

```python

is_example_notebook_like = len(script_blocks) > 2

for blabel, bcontent in script_blocks:
    ...
    if has_example_notebook_style:
        ...
    else:
        ...
```

This way you can even get rid of the comments